core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -1617,7 +1615,7 @@ public void receivePending(Transaction tx, @Nullable List<Transaction> dependenc
             // If this tx spends any of our unspent outputs, mark them as spent now, then add to the pending pool. This
             // ensures that if some other client that has our keys broadcasts a spend we stay in sync. Also updates the
             // timestamp on the transaction and registers/runs event listeners.
-            commitTx(tx);
+            commitTx(tx, false);
```

Comment:
In our use case, there is a high transaction rate, so saving the wallet and forcing a sync every time is not performing well.  Would it be better to have this be configurable, with a default of saving now?

I can definitely see that you'd want to persist in the Bluetooth case.  In the other cases, your effect on the propagation of the tx is likely to be minimal, since you are just an SPV client with a small number of connections.